rscripts/YAL5-codon-pairs.R

```diff
@@ -0,0 +1,1455 @@
+#' extracted from transcript_info_tibble
+#' @param expand_width integer which provides the number of positions on each
+#' side of the feature of interest to include in the window
+#' @param size the size of the text on the metafeature plot
+#'
+#' @return A plot which shows the ribosomal occupancy around a feature of
+#' interest
+#'
+#' @example
+#'
+#' gff_df <- readGFFAsDf(here::here("..",
+#'                                  "example-datasets",
+#'                                  "simulated",
+#'                                  "mok",
+#'                                  "annotation",
+#'                                  "Scer_YAL_5genes_w_250utrs.gff3"))
+#'
+#' GeneratePlot(codon_pos_pair_i200,
+#'              gene_names,
+#'              dataset = "Mok-simYAL5",
+#'              hd_file = here::here("Mok-simYAL5", "output", "A", "A.h5"),
+#'              gff_df,
+#'              min_read_length = 10,
+#'              snapdisp = 0L,
+#'              asite_disp_path = here::here(
+#'                "data",
+#'                "yeast_standard_asite_disp_length.txt"),
+#'              filter_for_frame = TRUE,
+#'              feature_of_interest = "TCC AAG",
+#'              expand_width = 5L,
+#'              size = 12)
+#'
+#' @export
+GeneratePlot <- function(codon_pos_pair_i200, gene_names, dataset, hd_file,
+                         gff_df, min_read_length, snapdisp, asite_disp_path,
+                         filter_for_frame, feature_of_interest,
+                         expand_width, size = 12) {
+
+  overlayed_tibbles <- OverlayedTibble(codon_pos_pair_i200,
+                                       gene_names, dataset, hd_file,
+                                       gff_df, min_read_length,
+                                       snapdisp, asite_disp_path,
+                                       filter_for_frame, feature_of_interest,
+                                       expand_width)
+
+  overlayed_plot <- ggplot(overlayed_tibbles,
+                           mapping = aes(x = RelPos,
+                                         y = RelCount)) +
+    geom_line() +
+    theme_bw() +
+    theme(text = element_text(size = size),
+          axis.title = element_text(size = size, face = "bold"),
+          title = element_text(size = size, face = "bold")) +
+    labs(title = paste0("Meta-feature plot of codon pair ",
+                        feature_of_interest),
+         x = "Distance from codon pair (3nt)",
+         y = "Normalised ribosomal occupancy", size = 2) +
+    scale_x_continuous(breaks = seq(-expand_width, expand_width, 1))
+}
+#TEST: GeneratePlot(): produces a single plot =TRUE
+#TEST: GeneratePlot(): title is "Meta-feature plot of codon pair
+#      <feature_of_interest>
+#Test: GeneratePlot(): the x-axis is "Distance from codon pair (3nt) and y-axis
+#      is "Normalised ribosomal occupancy"
+
+ploot <- GeneratePlot(codon_pos_pair_i200, gene_names, dataset, hd_file,
```

Comment:
ploot -> plot
However this command seems to be repeated in line 1234 so maybe delete?

Reply:
Deleted